source/common/common/fancy_logger.cc

```diff
@@ -90,6 +90,15 @@ void FancyContext::setAllFancyLoggers(spdlog::level::level_enum level)
   }
 }
 
+FancyLogLevelMap FancyContext::getAllFancyLogLevels() ABSL_LOCKS_EXCLUDED(fancy_log_lock_) {
```

Comment:
I lack some context here, but if this is a production code utility, would this deserve a unit test of its own, and maybe add caching to it? 
If not, would it be worth considering moving this into test/ as right now it's only used there? 
I don't want to bloat this PR, just curious.

Reply:
fancy_logger.h comments say this is test only, I don't mind renaming to `getAllFancyLogLevelsForTest()` if that makes it clearer. Not sure how to move it out of this file as it needs access to internal fields.

Reply:
Yes, please name test-only code as such. Makes it much easier to understand.